src/gradualizer_db.erl

```diff
@@ -21,6 +27,9 @@
 %% Types for the Erlang Abstract Format
 -type type() :: erl_parse:abstract_type().
 
+%% Compiled regular expression
+-type regexp() :: {re_pattern, _, _, _, _}.
```

Comment:
I couldn't understand why you didn't want to use `re:mp()`, but now I see: It is not exported! So I understand now.

    5> gradualizer_db:get_exported_type(re,mp,[]).    
    not_exported

(That's annoying as from the docs it looks like it is the type to use. Perhaps I will do a PR at erlang/otp and export this type.)